feat: DH-19501: Expose Aggregated Columns in JSTreeTable

vbabich opened the pull request.

niloc132 reviewed (commented):
Looks like it should be correct to me - add some integration tests to confirm? With constituents and without, with unagg'd columns left in when the rollup is made, etc

vbabich commented:
> Looks like it should be correct to me - add some integration tests to confirm? With constituents and without, with unagg'd columns left in when the rollup is made, etc

Added unit tests, found and fixed the issue where `aggregatedColumns` were populated in non-rollup tree tables.
